### ISSUE TITLE
Unable to connect to Datastore - Text Detector 

### ISSUE THREAD

**Issue by swathionchip** [hellohaptik/chatbot_ner#269]
Hi All,

I wrote sample example for "**Detecting text entity from message**". I am getting below error.

_File "../datastore/datastore.py", line 48, in __init__
    raise DataStoreSettingsImproperlyConfiguredException()
datastore.exceptions.DataStoreSettingsImproperlyConfiguredException: 'Chatbot NER datastore settings are not configured correctly. Please make sure the required connection settings are available in the environment variables_

How to resolve this Issue.
Thanks In advance.

Regards,
Swathi.

**Comment by chiragjn** [comment]
Hello, sorry for the late reply, but can you please post the steps you followed for us to be able to recreate and help you?

Chances are most likely it is an issue with installation

**Comment by prasannaonchip** [comment]
step 1:
we Downloaded  the source code from "https://github.com/hellohaptik/chatbot_ner".
And we Renamed  the file named as config.example to config and after that we updated  the variable  ES_URL with "ES_URL=http://localhost:9200"
step 2:
After that we created the one virtual environment for the python 2 version by using the producer as followed
sudo apt-get install virtualenv
virtualenv -p python ~/my_python2_environment
source ~/my_python2_environment/bin/activate
step 3:

Install requirements.txt which are present in the source code of hellohaptik chatbotner.
pip install -r requirements.txt

step 4:
wget get.docker.com
bash index.html
sudo usermod -aG docker $USER

step 5:
sudo curl -L https://github.com/docker/compose/releases/download/1.22.0/docker-compose-$(uname -s)-$(uname -m) -o /usr/local/bin/docker-compose
sudo chmod +x /usr/local/bin/docker-compose

step 6:
cd chatbot_ner 
cp config.example .env  
cp .env docker/.env
cd docker
docker-compose up --build -d

step 7:
sudo lsof -t -i tcp:8081 -s tcp:listen | sudo xargs kill

step 8:
cd ~/chatbot_ner/docker 
docker-compose ps 
docker exec -it container-name bash

step 8:
$ cd ~/chatbot_ner/
$ python manage.py shell

step 9 :
here we tested the example which are mentioned in the document named adding_entities.md
from datastore import DataStore
csv_file = '~/attachment_types.csv' # example file path to the csv file
db = DataStore()
db.populate(csv_file_paths=[csv_file,])

while runing this we are getting the error is 

EngineConnectionException: 'Chatbot NER datastore was unable to connect to elasticsearch. Please make sure the elasticsearch service is reachable.'
 
and also text  api example are also not working 

Example 1: Detecting text entity from message

Django Shell:

message=u'i want to order chinese from mainland china and pizza from dominos'
entity_name='restaurant' # here detection NER will search for values in dictionary 'restaurant'  
structured_value=None
fallback_value=None
bot_message=None
source_language='en'

from ner_v1.chatbot.entity_detection import get_text
output = get_text(message=message, entity_name=entity_name,
                  structured_value=structured_value,
                  fallback_value=fallback_value,
                  bot_message=bot_message,language=source_language)
print(output)

here also we are getting the same error like

EngineConnectionException: 'Chatbot NER datastore was unable to connect to elasticsearch. Please make sure the elasticsearch service is reachable.'
 
please reply fast

thanks in advance.

**Comment by chiragjn** [comment]
Hello, step 1-3 that set up the virtual environment were not necessary. It's alright as no harm was done. I think the problem happened because you changed the ES_HOST variable

try this (first, make sure you are outside the docker)

```
cd chatbot_ner/docker
vim .env # Or use any editor you like
```

Make sure all ES related entries look like follows, especially `ES_HOST` should be `elasticsearch`  because our docker-compose makes sure to bring up an elasticsearch container (unless you want to have your own elasticsearch hosted somewhere)

```
ENGINE=elasticsearch
ES_AUTH_NAME=
ES_AUTH_PASSWORD=
ES_HOST=elasticsearch
ES_URL=
ES_PORT=9200
ES_INDEX_NAME=entity_data
ES_DOC_TYPE=data_dictionary
```

After this just restart the container by running the following in the `docker/` folder

```
docker-compose down
docker-compose up -d
```

**Comment by prasannaonchip** [comment]
in pace of ES_URL  i used the  ES_URL=http://localhost:9200 in configuration file.
is this ok  or not

**Comment by chiragjn** [comment]
If you setup Chatbot NER using docker, we include the elasticsearch container with our main container, so you don't need to set up your own elasticsearch. That is why the `ES_HOST` should be `elasticsearch`

**Comment by prasannaonchip** [comment]
i followed the producer what you mentioned in the above,but this time also i am getting the same error 
like

**Comment by prasannaonchip** [comment]
now also text detector not working it is showing the same issue 
EngineConnectionException: 'Chatbot NER datastore was unable to connect to elasticsearch. Please make sure the elasticsearch service is reachable.'
while running in the django shell.
thanks in advance

**Comment by chiragjn** [comment]
Addressed in #268 
https://github.com/hellohaptik/chatbot_ner/issues/268#issuecomment-507138365
